crates/astria-sequencer/src/proposal/commitment.rs

```diff
@@ -0,0 +1,220 @@
+        let sequence_action = Action::SequenceAction(sequence::Action::new(
+            b"testchainid".to_vec(),
+            b"helloworld".to_vec(),
+        ));
+
+        let signing_key = SigningKey::new(OsRng);
+        let tx = Unsigned {
+            nonce: Nonce::from(0),
+            actions: vec![
+                sequence_action.clone(),
+                Action::TransferAction(accounts::Transfer::new(
+                    Address([0u8; 20]),
+                    Balance::from(1),
+                )),
+            ],
+        };
+
+        let signed_tx = tx.into_signed(&signing_key);
+        let tx_bytes = signed_tx.to_proto().encode_to_vec();
+        let txs = vec![tx_bytes.into()];
+        let (action_commitment_0, _) = generate_sequence_actions_commitment(txs);
+
+        let signing_key = SigningKey::new(OsRng);
+        let tx = Unsigned {
+            nonce: Nonce::from(0),
+            actions: vec![sequence_action],
+        };
+
+        let signed_tx = tx.into_signed(&signing_key);
+        let tx_bytes = signed_tx.to_proto().encode_to_vec();
+        let txs = vec![tx_bytes.into()];
+        let (action_commitment_1, _) = generate_sequence_actions_commitment(txs);
+        assert_eq!(action_commitment_0, action_commitment_1);
+    }
+
+    #[test]
+    fn generate_action_tree_leaves_assert_leaves_ordered_by_chain_id() {
+        let signing_key = SigningKey::new(OsRng);
+
+        let chain_id_0 = b"testchainid0";
+        let tx = Unsigned {
+            nonce: Nonce::from(0),
+            actions: vec![Action::SequenceAction(sequence::Action::new(
+                chain_id_0.to_vec(),
+                b"helloworld".to_vec(),
+            ))],
+        };
+        let signed_tx_0 = tx.into_signed(&signing_key);
+
+        let chain_id_1 = b"testchainid1";
+        let tx = Unsigned {
+            nonce: Nonce::from(0),
+            actions: vec![Action::SequenceAction(sequence::Action::new(
+                chain_id_1.to_vec(),
+                b"helloworld".to_vec(),
+            ))],
+        };
+        let signed_tx_1 = tx.into_signed(&signing_key);
+
+        let chain_id_2 = b"testchainid2";
+        let tx = Unsigned {
+            nonce: Nonce::from(0),
+            actions: vec![Action::SequenceAction(sequence::Action::new(
+                chain_id_2.to_vec(),
+                b"helloworld".to_vec(),
+            ))],
+        };
+        let signed_tx_2 = tx.into_signed(&signing_key);
+
+        let txs = vec![signed_tx_0, signed_tx_1, signed_tx_2];
+        let chain_id_to_txs = group_sequence_actions_by_chain_id(&txs);
+        let leaves = generate_action_tree_leaves(chain_id_to_txs);
+        leaves.iter().enumerate().for_each(|(i, leaf)| {
+            if i == 0 {
+                return;
+            }
+            assert!(leaf > &leaves[i - 1]);
+        });
+    }
+
+    #[test]
+    fn generate_sequence_actions_commitment_snapshot() {
```

Comment:
Is there are a more descriptive name for this test?

Reply:
I'll think of one + add a comment